docs/rate_limiter.md

```diff
@@ -0,0 +1,131 @@
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Rate Limiter
+
+Rate limiter manages the rate at which requests are scheduled on
+model instances by Triton. The rate limiter operates across all
+models loaded in Triton to allow *cross-model prioritization*.
+
+In absence of rate limiting (--rate-limit=off), Triton schedules
+execution of a request (or set of requests when using dynamic
+batching) as soon as a model instance is available. This behavior
+is typically best suited for performance. However, there can be
+cases where running all the models simultaneously places excessive
+load on the server. For instance, model execution on some
+frameworks dynamically allocate memory. Running all such models
+simultaneously may lead to system going out-of-memory.
+
+Rate limiter allows to post-pone the inference execution on some
+model instances such that not all of them runs simultaneously. 
+The model priorities are used to decide which model instance
+to schedule next. 
+
+## Using Rate Limiter
+
+To enable rate limiting users must set `--rate-limit` option when
+launching tritonserver. For more information, consult usage of
+the option emitted by `tritonserver --help`.
+
+The rate limiter is controlled by rate limiter configuration
+described in [rate limiter config](model_configuration.md#rate-limiter-config).
+The rate limiter configuration includes 
+[resources](model_configuration.md#resources) and
+[priority](model_configuration.md#priority) for the instances
+defined by the instance group.
+
+### Resources
+
+Resources are identified by a unique name and a count indicating
+the number of copies of the resource. By default, model instance
+uses no rate-limiter resources. By listing a resource/count the
+model instance indicates that it requires that many resources to
+be available on the model instance device before it can be allowed
+to execute. When under execution the specified many resources are
+allocated to the model instance only to be released when the
+execution is over. The available number of resource copies
+are, by default, the max across all model instances that list that
+resource (and so must be adjusted each time a model instance is
```

Comment:
I would remove the parenthetic statement. It makes it seem like the user needs to do something when models are added/removed and it is not clear what they must do